contracts/SwapFeeManager.sol

```diff
@@ -0,0 +1,76 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.20;
+import "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+import "@openzeppelin/contracts/access/Ownable.sol";
+
+contract SwapFeeManager is Ownable {
+    using SafeERC20 for IERC20;
+
+    address public immutable dexFeeWallet;
+    address public immutable burnFeeWallet;
+
+    uint256 private constant BURN_FEE_PERCENT = 25;
+    uint256 private constant TOTAL_PERCENT = 100;
+
+    event FeesSplit(uint256 dexFeeAmount, uint256 burnFeeAmount);
+
+    constructor(
+        address _dexFeeWallet,
+        address _burnFeeWallet
+    ) Ownable(_msgSender()) {
+        require(
+            _dexFeeWallet != address(0),
+            "dexFeeWallet must not be zero address"
+        );
+        require(
+            _burnFeeWallet != address(0),
+            "burnFeeWallet must not be zero address"
+        );
+
+        dexFeeWallet = _dexFeeWallet;
+        burnFeeWallet = _burnFeeWallet;
+    }
+
+    // Function to receive Ether
+    receive() external payable {}
+
+    /**
+     * @dev Splits the Ether balance in the contract into 75% for dexFeeWallet and 25% for burnFeeWallet.
+     */
+    function splitAndWithdraw() external onlyOwner {
```

Comment:
I would suggest to allow anyone to split and withdraw, doesn't have to be the contract deployer.

Reply:
I recommend keeping onlyOwner for the dust/griefing protection reason.

Without access control, malicious actors could repeatedly call splitAndWithdraw() with minimal amounts on contract address, forcing recipient wallets to accumulate dust - small amounts that cost more to transfer than they're worth.

Alternatively we could add map of additional authorised addresses, which also could call split function. So inside splitAndWithdraw we would call canSplit check
smth like this:

```solidity
modifier canSplit() {
    require(msg.sender == owner() || splitters[msg.sender], "Not authorized");
```

However, this approach adds extra storage costs, execution costs (we have to read from storage).

Its better to keep onlyOwner for spam bots protection.

Reply:
can't we let those bots pay the transfer (contract execution) fees?

Reply:
Ah, yes you are right, that as msg sender is actually pays fees, then spam bots will do it.
Ok, I will remove owner. But please remember about the owner-only access option.

Reply:
removed owner access restriction, updated tests https://github.com/KomodoPlatform/etomic-swap/pull/8/commits/d8ad7f68d7351993170d3f2dbc331bc24fac9280